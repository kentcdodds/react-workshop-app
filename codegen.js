module.exports = require('./dist/codegen').getCode
