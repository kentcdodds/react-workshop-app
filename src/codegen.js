// this generates the code to use in the entry file

const loadFiles = require('./load-files')

function getCode({cwd = process.cwd(), ignore, renderOptions} = {}) {
  const filesInfo = loadFiles({cwd, ignore})

  const imports = filesInfo.map(({id, fullFilePath, ext}) => {
    let loaders = ''
    if (ext === '.html') {
      loaders = '!raw-loader!'
    } else if (ext === '.md' || ext === '.mdx') {
      loaders = '!babel-loader!mdx-loader!'
    }
    return `"${id}": () => import("${loaders}${fullFilePath}")`
  })

  return `
import makeWorkshopApp from '@kentcdodds/react-workshop-app'
import pkg from '../package.json'

const filesInfo = ${JSON.stringify(filesInfo, null, 2)}

makeWorkshopApp({
  imports: {
    ${imports.join(',\n    ')}
  },
  filesInfo,
  projectTitle: pkg.title,
  ${renderOptions ? `renderOptions: ${JSON.stringify(renderOptions)}` : ''}
})`.trim()
}

module.exports = getCode
