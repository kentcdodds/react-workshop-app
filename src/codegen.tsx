// this generates the code to use in the entry file
import {getAppInfo} from './get-app-info'

function getCode({
  cwd = process.cwd(),
  ignore,
  options,
}: {
  cwd?: string
  ignore?: Array<string>
  options?: Record<string, unknown>
} = {}) {
  const {gitHubRepoUrl, filesInfo, hasBackend, imports} = getAppInfo({
    cwd,
    ignore,
  })
  return `
import {makeKCDWorkshopApp} from '@kentcdodds/react-workshop-app'
import {loadDevTools} from '@kentcdodds/react-workshop-app/dev-tools'
import pkg from '../package.json'
${hasBackend ? `import * as backend from './backend'` : ''}

if (module.hot) module.hot.accept()

const filesInfo = ${JSON.stringify(filesInfo, null, 2)}

loadDevTools(() => {
  makeKCDWorkshopApp({
    imports: {
      ${imports.join(',\n      ')}
    },
    filesInfo,
    projectTitle: pkg.title,
    gitHubRepoUrl: \`${gitHubRepoUrl}\`,
    ${hasBackend ? `backend,` : ''}
    ${options ? `options: ${JSON.stringify(options)},` : ''}
  })
})`.trim()
}

export {getCode}
