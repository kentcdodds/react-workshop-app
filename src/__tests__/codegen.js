import path from 'path'
import codegen from '../codegen'

const cwd = path.join(__dirname, '../../example/react-fundamentals')

expect.addSnapshotSerializer({
  test(val) {
    return typeof val === 'string'
  },
  print(val) {
    return val.split(cwd).join('<PROJECT_ROOT>')
  },
})

test('review the snapshot please', () => {
  expect(codegen({cwd})).toMatchSnapshot()
})
