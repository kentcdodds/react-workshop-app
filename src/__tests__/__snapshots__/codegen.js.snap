// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`review the snapshot please 1`] = `
import makeWorkshopApp from '@kentcdodds/react-workshop-app'
import pkg from '../package.json'

const filesInfo = [
  {
    "id": "src/examples/example.js",
    "fullFilePath": "<PROJECT_ROOT>/src/examples/example.js",
    "filePath": "src/examples/example.js",
    "isolatedPath": "/isolated/examples/example.js",
    "ext": ".js",
    "filename": "example",
    "type": "examples",
    "number": 0,
    "extraCreditNumber": 0
  },
  {
    "id": "src/exercise/01.html",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/01.html",
    "filePath": "src/exercise/01.html",
    "isolatedPath": "/isolated/exercise/01.html",
    "ext": ".html",
    "filename": "01",
    "type": "exercise",
    "number": 1,
    "extraCreditNumber": 1,
    "title": "Basic JavaScript-rendered Hello World",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/01.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/01.md",
    "filePath": "src/exercise/01.md",
    "isolatedPath": "/isolated/exercise/01.md",
    "ext": ".md",
    "filename": "01",
    "type": "instruction",
    "number": 1,
    "extraCreditNumber": 1,
    "title": "Basic JavaScript-rendered Hello World"
  },
  {
    "id": "src/exercise/02.html",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/02.html",
    "filePath": "src/exercise/02.html",
    "isolatedPath": "/isolated/exercise/02.html",
    "ext": ".html",
    "filename": "02",
    "type": "exercise",
    "number": 2,
    "extraCreditNumber": 2,
    "title": "Intro to raw React APIs",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/02.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/02.md",
    "filePath": "src/exercise/02.md",
    "isolatedPath": "/isolated/exercise/02.md",
    "ext": ".md",
    "filename": "02",
    "type": "instruction",
    "number": 2,
    "extraCreditNumber": 2,
    "title": "Intro to raw React APIs"
  },
  {
    "id": "src/exercise/03.html",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/03.html",
    "filePath": "src/exercise/03.html",
    "isolatedPath": "/isolated/exercise/03.html",
    "ext": ".html",
    "filename": "03",
    "type": "exercise",
    "number": 3,
    "extraCreditNumber": 3,
    "title": "Using JSX",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/03.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/03.md",
    "filePath": "src/exercise/03.md",
    "isolatedPath": "/isolated/exercise/03.md",
    "ext": ".md",
    "filename": "03",
    "type": "instruction",
    "number": 3,
    "extraCreditNumber": 3,
    "title": "Using JSX"
  },
  {
    "id": "src/exercise/04.html",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/04.html",
    "filePath": "src/exercise/04.html",
    "isolatedPath": "/isolated/exercise/04.html",
    "ext": ".html",
    "filename": "04",
    "type": "exercise",
    "number": 4,
    "extraCreditNumber": 4,
    "title": "Creating custom components",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/04.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/04.md",
    "filePath": "src/exercise/04.md",
    "isolatedPath": "/isolated/exercise/04.md",
    "ext": ".md",
    "filename": "04",
    "type": "instruction",
    "number": 4,
    "extraCreditNumber": 4,
    "title": "Creating custom components"
  },
  {
    "id": "src/exercise/05.js",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/05.js",
    "filePath": "src/exercise/05.js",
    "isolatedPath": "/isolated/exercise/05.js",
    "ext": ".js",
    "filename": "05",
    "type": "exercise",
    "number": 5,
    "extraCreditNumber": 5,
    "title": "Styling",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/05.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/05.md",
    "filePath": "src/exercise/05.md",
    "isolatedPath": "/isolated/exercise/05.md",
    "ext": ".md",
    "filename": "05",
    "type": "instruction",
    "number": 5,
    "extraCreditNumber": 5,
    "title": "Styling"
  },
  {
    "id": "src/exercise/06.js",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/06.js",
    "filePath": "src/exercise/06.js",
    "isolatedPath": "/isolated/exercise/06.js",
    "ext": ".js",
    "filename": "06",
    "type": "exercise",
    "number": 6,
    "extraCreditNumber": 6,
    "title": "Basic Forms",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/06.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/06.md",
    "filePath": "src/exercise/06.md",
    "isolatedPath": "/isolated/exercise/06.md",
    "ext": ".md",
    "filename": "06",
    "type": "instruction",
    "number": 6,
    "extraCreditNumber": 6,
    "title": "Basic Forms"
  },
  {
    "id": "src/exercise/07.js",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/07.js",
    "filePath": "src/exercise/07.js",
    "isolatedPath": "/isolated/exercise/07.js",
    "ext": ".js",
    "filename": "07",
    "type": "exercise",
    "number": 7,
    "extraCreditNumber": 7,
    "title": "Dynamic Forms",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/07.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/07.md",
    "filePath": "src/exercise/07.md",
    "isolatedPath": "/isolated/exercise/07.md",
    "ext": ".md",
    "filename": "07",
    "type": "instruction",
    "number": 7,
    "extraCreditNumber": 7,
    "title": "Dynamic Forms"
  },
  {
    "id": "src/exercise/08.js",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/08.js",
    "filePath": "src/exercise/08.js",
    "isolatedPath": "/isolated/exercise/08.js",
    "ext": ".js",
    "filename": "08",
    "type": "exercise",
    "number": 8,
    "extraCreditNumber": 8,
    "title": "Controlled Forms",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/08.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/08.md",
    "filePath": "src/exercise/08.md",
    "isolatedPath": "/isolated/exercise/08.md",
    "ext": ".md",
    "filename": "08",
    "type": "instruction",
    "number": 8,
    "extraCreditNumber": 8,
    "title": "Controlled Forms"
  },
  {
    "id": "src/exercise/09.js",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/09.js",
    "filePath": "src/exercise/09.js",
    "isolatedPath": "/isolated/exercise/09.js",
    "ext": ".js",
    "filename": "09",
    "type": "exercise",
    "number": 9,
    "extraCreditNumber": 9,
    "title": "Rendering Lists",
    "extraCreditTitle": ""
  },
  {
    "id": "src/exercise/09.md",
    "fullFilePath": "<PROJECT_ROOT>/src/exercise/09.md",
    "filePath": "src/exercise/09.md",
    "isolatedPath": "/isolated/exercise/09.md",
    "ext": ".md",
    "filename": "09",
    "type": "instruction",
    "number": 9,
    "extraCreditNumber": 9,
    "title": "Rendering Arrays"
  },
  {
    "id": "src/final/01.extra-1.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/01.extra-1.html",
    "filePath": "src/final/01.extra-1.html",
    "isolatedPath": "/isolated/final/01.extra-1.html",
    "ext": ".html",
    "filename": "01.extra-1",
    "type": "extraCredit",
    "number": 1,
    "extraCreditNumber": 1,
    "title": "Basic JavaScript-rendered Hello World",
    "extraCreditTitle": "generate the root node"
  },
  {
    "id": "src/final/01.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/01.html",
    "filePath": "src/final/01.html",
    "isolatedPath": "/isolated/final/01.html",
    "ext": ".html",
    "filename": "01",
    "type": "final",
    "number": 1,
    "extraCreditNumber": 1,
    "title": "Basic JavaScript-rendered Hello World",
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/02.extra-1.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/02.extra-1.html",
    "filePath": "src/final/02.extra-1.html",
    "isolatedPath": "/isolated/final/02.extra-1.html",
    "ext": ".html",
    "filename": "02.extra-1",
    "type": "extraCredit",
    "number": 2,
    "extraCreditNumber": 1,
    "title": "Intro to raw React APIs",
    "extraCreditTitle": "nesting elements"
  },
  {
    "id": "src/final/02.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/02.html",
    "filePath": "src/final/02.html",
    "isolatedPath": "/isolated/final/02.html",
    "ext": ".html",
    "filename": "02",
    "type": "final",
    "number": 2,
    "extraCreditNumber": 2,
    "title": "Intro to raw React APIs",
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/03.extra-1.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/03.extra-1.html",
    "filePath": "src/final/03.extra-1.html",
    "isolatedPath": "/isolated/final/03.extra-1.html",
    "ext": ".html",
    "filename": "03.extra-1",
    "type": "extraCredit",
    "number": 3,
    "extraCreditNumber": 1,
    "title": "Using JSX",
    "extraCreditTitle": "interpolate className and children"
  },
  {
    "id": "src/final/03.extra-2.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/03.extra-2.html",
    "filePath": "src/final/03.extra-2.html",
    "isolatedPath": "/isolated/final/03.extra-2.html",
    "ext": ".html",
    "filename": "03.extra-2",
    "type": "extraCredit",
    "number": 3,
    "extraCreditNumber": 2,
    "title": "Using JSX",
    "extraCreditTitle": "spread props"
  },
  {
    "id": "src/final/03.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/03.html",
    "filePath": "src/final/03.html",
    "isolatedPath": "/isolated/final/03.html",
    "ext": ".html",
    "filename": "03",
    "type": "final",
    "number": 3,
    "extraCreditNumber": 3,
    "title": "Using JSX",
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/04.extra-1.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/04.extra-1.html",
    "filePath": "src/final/04.extra-1.html",
    "isolatedPath": "/isolated/final/04.extra-1.html",
    "ext": ".html",
    "filename": "04.extra-1",
    "type": "extraCredit",
    "number": 4,
    "extraCreditNumber": 1,
    "title": "Creating custom components",
    "extraCreditTitle": "using a custom component with React.createElement"
  },
  {
    "id": "src/final/04.extra-2.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/04.extra-2.html",
    "filePath": "src/final/04.extra-2.html",
    "isolatedPath": "/isolated/final/04.extra-2.html",
    "ext": ".html",
    "filename": "04.extra-2",
    "type": "extraCredit",
    "number": 4,
    "extraCreditNumber": 2,
    "title": "Creating custom components",
    "extraCreditTitle": "using a custom component with JSX"
  },
  {
    "id": "src/final/04.html",
    "fullFilePath": "<PROJECT_ROOT>/src/final/04.html",
    "filePath": "src/final/04.html",
    "isolatedPath": "/isolated/final/04.html",
    "ext": ".html",
    "filename": "04",
    "type": "final",
    "number": 4,
    "extraCreditNumber": 4,
    "title": "Creating custom components",
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/05.extra-1.js",
    "fullFilePath": "<PROJECT_ROOT>/src/final/05.extra-1.js",
    "filePath": "src/final/05.extra-1.js",
    "isolatedPath": "/isolated/final/05.extra-1.js",
    "ext": ".js",
    "filename": "05.extra-1",
    "type": "extraCredit",
    "number": 5,
    "extraCreditNumber": 1,
    "title": "Styling",
    "extraCreditTitle": "Custom component that encapsulates styling"
  },
  {
    "id": "src/final/05.js",
    "fullFilePath": "<PROJECT_ROOT>/src/final/05.js",
    "filePath": "src/final/05.js",
    "isolatedPath": "/isolated/final/05.js",
    "ext": ".js",
    "filename": "05",
    "type": "final",
    "number": 5,
    "extraCreditNumber": 5,
    "title": "Styling",
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/06.extra-1.js",
    "fullFilePath": "<PROJECT_ROOT>/src/final/06.extra-1.js",
    "filePath": "src/final/06.extra-1.js",
    "isolatedPath": "/isolated/final/06.extra-1.js",
    "ext": ".js",
    "filename": "06.extra-1",
    "type": "extraCredit",
    "number": 6,
    "extraCreditNumber": 1,
    "title": "Basic Forms",
    "extraCreditTitle": "using refs"
  },
  {
    "id": "src/final/06.js",
    "fullFilePath": "<PROJECT_ROOT>/src/final/06.js",
    "filePath": "src/final/06.js",
    "isolatedPath": "/isolated/final/06.js",
    "ext": ".js",
    "filename": "06",
    "type": "final",
    "number": 6,
    "extraCreditNumber": 6,
    "title": "Basic Forms",
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/07.js",
    "fullFilePath": "<PROJECT_ROOT>/src/final/07.js",
    "filePath": "src/final/07.js",
    "isolatedPath": "/isolated/final/07.js",
    "ext": ".js",
    "filename": "07",
    "type": "final",
    "number": 7,
    "extraCreditNumber": 7,
    "title": "Dynamic Forms",
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/08.js",
    "fullFilePath": "<PROJECT_ROOT>/src/final/08.js",
    "filePath": "src/final/08.js",
    "isolatedPath": "/isolated/final/08.js",
    "ext": ".js",
    "filename": "08",
    "type": "final",
    "number": 8,
    "extraCreditNumber": 8,
    "title": "Controlled Forms",
    "extraCreditTitle": ""
  },
  {
    "id": "src/final/09.extra-1.js",
    "fullFilePath": "<PROJECT_ROOT>/src/final/09.extra-1.js",
    "filePath": "src/final/09.extra-1.js",
    "isolatedPath": "/isolated/final/09.extra-1.js",
    "ext": ".js",
    "filename": "09.extra-1",
    "type": "extraCredit",
    "number": 9,
    "extraCreditNumber": 1,
    "title": "Rendering Lists",
    "extraCreditTitle": "Focus Demo"
  },
  {
    "id": "src/final/09.js",
    "fullFilePath": "<PROJECT_ROOT>/src/final/09.js",
    "filePath": "src/final/09.js",
    "isolatedPath": "/isolated/final/09.js",
    "ext": ".js",
    "filename": "09",
    "type": "final",
    "number": 9,
    "extraCreditNumber": 9,
    "title": "Rendering Lists",
    "extraCreditTitle": ""
  },
  {
    "id": "src/index.js",
    "fullFilePath": "<PROJECT_ROOT>/src/index.js",
    "filePath": "src/index.js",
    "isolatedPath": "/isolated/index.js",
    "ext": ".js",
    "filename": "index",
    "type": "src",
    "number": 0,
    "extraCreditNumber": 0
  },
  {
    "id": "src/server.js",
    "fullFilePath": "<PROJECT_ROOT>/src/server.js",
    "filePath": "src/server.js",
    "isolatedPath": "/isolated/server.js",
    "ext": ".js",
    "filename": "server",
    "type": "src",
    "number": 0,
    "extraCreditNumber": 0
  },
  {
    "id": "src/setupTests.js",
    "fullFilePath": "<PROJECT_ROOT>/src/setupTests.js",
    "filePath": "src/setupTests.js",
    "isolatedPath": "/isolated/setupTests.js",
    "ext": ".js",
    "filename": "setupTests",
    "type": "src",
    "number": 0,
    "extraCreditNumber": 0
  },
  {
    "id": "src/styles.css",
    "fullFilePath": "<PROJECT_ROOT>/src/styles.css",
    "filePath": "src/styles.css",
    "isolatedPath": "/isolated/styles.css",
    "ext": ".css",
    "filename": "styles",
    "type": "src",
    "number": 0,
    "extraCreditNumber": 0
  }
]

makeWorkshopApp({
  imports: {
    "src/examples/example.js": () => import("<PROJECT_ROOT>/src/examples/example.js"),
    "src/exercise/01.html": () => import("!raw-loader!<PROJECT_ROOT>/src/exercise/01.html"),
    "src/exercise/01.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/01.md"),
    "src/exercise/02.html": () => import("!raw-loader!<PROJECT_ROOT>/src/exercise/02.html"),
    "src/exercise/02.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/02.md"),
    "src/exercise/03.html": () => import("!raw-loader!<PROJECT_ROOT>/src/exercise/03.html"),
    "src/exercise/03.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/03.md"),
    "src/exercise/04.html": () => import("!raw-loader!<PROJECT_ROOT>/src/exercise/04.html"),
    "src/exercise/04.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/04.md"),
    "src/exercise/05.js": () => import("<PROJECT_ROOT>/src/exercise/05.js"),
    "src/exercise/05.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/05.md"),
    "src/exercise/06.js": () => import("<PROJECT_ROOT>/src/exercise/06.js"),
    "src/exercise/06.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/06.md"),
    "src/exercise/07.js": () => import("<PROJECT_ROOT>/src/exercise/07.js"),
    "src/exercise/07.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/07.md"),
    "src/exercise/08.js": () => import("<PROJECT_ROOT>/src/exercise/08.js"),
    "src/exercise/08.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/08.md"),
    "src/exercise/09.js": () => import("<PROJECT_ROOT>/src/exercise/09.js"),
    "src/exercise/09.md": () => import("!babel-loader!mdx-loader!<PROJECT_ROOT>/src/exercise/09.md"),
    "src/final/01.extra-1.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/01.extra-1.html"),
    "src/final/01.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/01.html"),
    "src/final/02.extra-1.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/02.extra-1.html"),
    "src/final/02.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/02.html"),
    "src/final/03.extra-1.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/03.extra-1.html"),
    "src/final/03.extra-2.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/03.extra-2.html"),
    "src/final/03.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/03.html"),
    "src/final/04.extra-1.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/04.extra-1.html"),
    "src/final/04.extra-2.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/04.extra-2.html"),
    "src/final/04.html": () => import("!raw-loader!<PROJECT_ROOT>/src/final/04.html"),
    "src/final/05.extra-1.js": () => import("<PROJECT_ROOT>/src/final/05.extra-1.js"),
    "src/final/05.js": () => import("<PROJECT_ROOT>/src/final/05.js"),
    "src/final/06.extra-1.js": () => import("<PROJECT_ROOT>/src/final/06.extra-1.js"),
    "src/final/06.js": () => import("<PROJECT_ROOT>/src/final/06.js"),
    "src/final/07.js": () => import("<PROJECT_ROOT>/src/final/07.js"),
    "src/final/08.js": () => import("<PROJECT_ROOT>/src/final/08.js"),
    "src/final/09.extra-1.js": () => import("<PROJECT_ROOT>/src/final/09.extra-1.js"),
    "src/final/09.js": () => import("<PROJECT_ROOT>/src/final/09.js"),
    "src/index.js": () => import("<PROJECT_ROOT>/src/index.js"),
    "src/server.js": () => import("<PROJECT_ROOT>/src/server.js"),
    "src/setupTests.js": () => import("<PROJECT_ROOT>/src/setupTests.js"),
    "src/styles.css": () => import("<PROJECT_ROOT>/src/styles.css")
  },
  filesInfo,
  projectTitle: pkg.title,
  
})
`;
